main.go

```diff
@@ -312,6 +312,14 @@ func main() {
 
 	if enabled, _ := features.Enabled(features.GitManagedTransport); enabled {
 		managed.InitManagedTransport(ctrl.Log.WithName("managed-transport"))
+	} else {
+		if optimize, _ := feathelper.Enabled(features.OptimizedGitClones); optimize {
+			setupLog.Error(
+				fmt.Errorf("OptimizedGitClones=true but GitManagedTransport=false"),
+				"git clones can only be optimized when using managed transort",
+			)
+			os.Exit(1)
+		}
```

Comment:
Given that `OptimizedGitClones` is enabled by default, I would probably just disable it and log a message stating that. Not sure it is worth erroring, as potentially this was not an active decision made by the user.